### ISSUE TITLE
Related to #5227

### ISSUE THREAD

**Issue by alb-i986** [travis-ci/travis-ci#6029]
Hi,

I've noticed something interesting, I think, which is probably related to issue #5227.

So, I've [submitted a PR](https://github.com/rholder/guava-retrying/pull/59) to the project guava-retrying, and came across #5227 as [the build failed
](https://travis-ci.org/rholder/guava-retrying/builds/128280354) with `buffer overflow detected ***: /usr/lib/jvm/java-7-openjdk-amd64/bin/java terminated`.

I have a travis account myself, and after activating the travis build for [my forked version of the project](https://github.com/alb-i986/guava-retrying), unexpectedly, it turns out that the builds are working for me!

In particular please compare these two builds:
- https://travis-ci.org/rholder/guava-retrying/builds/129006161
- https://travis-ci.org/alb-i986/guava-retrying/builds/129006158

They come from [the very same commit](https://github.com/alb-i986/guava-retrying/commit/cab052be731e5493a45668bd8b2860e50d945777), but the results are totally different.
The former was triggered by rholder's travis PR trigger, and it failed.
The latter was triggered by my Push trigger, and it passed.

In particular, look at the build for openjdk6:
- https://s3.amazonaws.com/archive.travis-ci.org/jobs/129006167/log.txt - rholder's

```
[0K[33;1mWorker information[0m
hostname: travis-worker-gce-org-prod5-2:c0d6415f-4e68-469b-a5fd-bc8eab8b44ac
version: v2.3.0 https://github.com/travis-ci/worker/tree/9c1321414ebd21739efdff448627af90c34e6610
```

- https://s3.amazonaws.com/archive.travis-ci.org/jobs/129006166/log.txt - mine

```
Using worker: worker-linux-docker-e5450d62.prod.travis-ci.org:travis-linux-10
```

The two builds seem to have run against two completely different build infrastructures.. Docker vs. pre-docker maybe? And the old infrastructure seems to use long hostnames, whereas the new one uses short hostnames. Which is what is making my builds pass and rholder's builds fail.
But why? what's the difference between rholder's travis and mine?

The only difference I can see is "PR build" vs. "push build". So I've tried submitting a PR to my own forked version, thus testing my "build PR" trigger, and [the result is that it's still passing](https://travis-ci.org/alb-i986/guava-retrying/builds/129005798). So it looks like it doesn't depend on PR vs. Push hook either.

Another difference is *when* we activated the travis build for the project. rholder activated it months ago, whereas I did it minutes ago. It feels like his travis project metadata was not migrated to the latest travis, or something like that, if you know what I mean, and that's why travis is sending his builds to the old infrastructure.

There's one last test I'd be curious to try.
@rholder should try the following:
- go to https://travis-ci.org/profile/
- disable travis for guava-retrying
- then re-enable it again

If I'm right, these steps should fix rholder's builds without the need for any [hostname addon](https://docs.travis-ci.com/user/hostname).

**Comment by rholder** [comment]
@alb-i986 I put a [workaround](https://github.com/rholder/esthree/commit/a4a5f9e37c2134faa9d7398564c1f3bc4208b8dd) in another one of my projects for a similar openjdk 6/7 error [here](https://travis-ci.org/rholder/esthree/jobs/103494475) to force it to use the newer infrastructure. I believe you're on the right track in assuming it's picking up new defaults for newer projects.

I'd be willing to try this out. I'll flip that on and off and kick off a build to inspect then update this ticket when that's completed. If that doesn't seem to fix the issue with new PR's coming in I'll put the `sudo` workaround in `master` anyway so we get working PR builds again.

**Comment by rholder** [comment]
Alright, I flipped it on and off and here's what happened https://travis-ci.org/rholder/guava-retrying/builds/69088605. Yay! My normal builds are broken, too :smile_cat:.

Thanks for opening this and letting me know PR's were getting weird. I'll go ahead and put that `sudo` workaround in master for this project, too. If there's a way for the Travis CI folks to mark a project as "use latest infrastructure" or some such I'd be open to clicking a button or testing something to get that on this project.

**Comment by BanzaiMan** [comment]
This is a duplicate of https://github.com/travis-ci/travis-ci/issues/5669. If the PR comes from a fork, the addon is somehow ineffective. I have not quite figured it out yet.

The reason that @rholder's and @alb-i986's builds run on different infrastructures is that they were created at different points in time. Older ones are routed to the "old" infrastructure, while the newer ones are routed to containers by default.
